server/broadcast.go

```diff
@@ -150,6 +157,8 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) {
 				cxn.needOrch <- struct{}{}
 			}
 			return
+		} else {
```

Comment:
The `else` is not really needed since there is a `return` above. This goes back to the earlier notes about minimizing code under conditionals and maximizing the straight-through code flow.